R/opts.R

```diff
@@ -74,7 +74,7 @@ generation_time_opts <- function(..., disease, source, max = 15L,
       dot_options$dist <- "gamma"
     }
     ## set max
-    if (!("dist" %in% names(dot_options))) {
+    if (!("max" %in% names(dot_options))) {
```

Comment:
don't think you need the if statement - `max` is a named argument so will never be in `names(dot_options)`

Reply:
It will when you supply a list which is what happens if you use `get_generation_time()` and then use `generation_time_opts()` on that distribution. That might be an interface to phase out but can see it being common as it stands.

Reply:
(as an aside I think ever allowing the user to specify a free list is kind of a bad idea - making them call `dist_spec` (i.e your dist fn) might be a good idea to enforce the right structure etc from the get go.

Reply:
Yes I was thinking similar given that I managed to confuse myself over my own code here. Could `trunc_opts`, `generation_time_opts` and `delay_opts` perhaps even eventually be axed and replaced by `dist_spec` calls (with a deprecation warning in this version)? From what I can see the only things preventing that are the backwards compatibility with specifying `disease` and `source` and the `prior_weight` option (which could perhaps go somewhere else or equally be scrapped).